flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/history/HistoryServerArchiveFetcher.java

```diff
@@ -221,17 +267,40 @@ public void run() {
 							}
 						}
 					}
-					if (numFetchedArchives > 0) {
-						updateJobOverview(webOverviewDir, webDir);
-						for (int x = 0; x < numFetchedArchives; x++) {
-							numArchivedJobs.countDown();
-						}
-					}
 				}
+
+				if (!jobsToRemove.isEmpty() && processArchiveDeletion) {
+					events.addAll(cleanupExpiredJobs(jobsToRemove));
+				}
+				if (!events.isEmpty()) {
+					updateJobOverview(webOverviewDir, webDir);
+				}
+				events.forEach(jobArchiveEventListener::accept);
 			} catch (Exception e) {
 				LOG.error("Critical failure while fetching/processing job archives.", e);
 			}
 		}
+
+		private List<ArchiveEvent> cleanupExpiredJobs(Set<String> jobsToRemove) {
+
+			List<ArchiveEvent> deleteOperations = new ArrayList<>();
+			LOG.info("Archive directories for jobs {} were deleted.", jobsToRemove);
+
+			cachedArchives.removeAll(jobsToRemove);
+			jobsToRemove.forEach(removedJobID -> {
+				try {
+					Files.deleteIfExists(new File(webOverviewDir, removedJobID + JSON_FILE_ENDING).toPath());
+					Files.deleteIfExists(new File(webJobDir, removedJobID + JSON_FILE_ENDING).toPath());
```

Comment:
redundant, the subsequent directory deletion takes care of this file.

Reply:
Removed, thanks.